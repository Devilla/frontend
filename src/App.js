import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
