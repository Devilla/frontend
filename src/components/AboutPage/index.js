import React, { Component } from 'react';
import {Animated} from "react-animated-css";
export default class About extends Component {
     componentDidMount(){
     window.scrollTo(0, 0)
   } 
  render() {
    return (
        <div>
            <div className="page-header"><div className="page-header-overlay">
            <div className="centered page-header-container w-container">
            <Animated className="page-header-title" animationIn="fadeInDown" animationOut="fadeOut" isVisible={true}>About us</Animated>
            </div></div></div>
            <div className="section innerpage">
                <div className="container w-container">
                    <p>We started Influence with one mission: to cut down the advertising cost and to convert more customers on your website.</p>
            <p>We use your existing customers’ data on your website and show it to your visitors so that they are socially influenced by buying behavior of people on the website.</p>
            <p>Using this tool we’ve been able to help a lot of businesses in increasing their money making and get more conversions.</p>
            <p>Social influence is the next big thing in marketing for influencing buyers to buy more.</p>
            <p>Let’s start with your free to start trial and get making more money for you.
            </p>
                   
                </div>
              </div>
        </div>
    );
  }
}
