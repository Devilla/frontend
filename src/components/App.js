import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
