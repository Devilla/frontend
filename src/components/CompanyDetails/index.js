import React, { Component } from 'react';
import { Animated } from "react-animated-css";
import { ToastContainer, toast } from 'react-toastify';
import { validateCompanyName, validateWebsite, submitCompanyDetails } from '../../services/FormUtils';
import { storeToken } from '../../services/Request';


const toastConfig = {
  position: toast.POSITION.BOTTOM_LEFT,
  autoClose: 2000
};

export default class CompanyDetails extends Component {
  constructor(props) {
    super(props);

    this.state = {
      companyName: '',
      website: ''
    };

    // Store the token passed in params. This will be used to continue process further
    storeToken(props.match.params.token);
  }

  // Allow empty field
  isCompanyNameValid = () => (this.state.companyName === '' || validateCompanyName(this.state.companyName));

  // Allow empty field
  isWebsiteValid = () => (this.state.website === '' || validateWebsite(this.state.website));

  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value});
  };

  handleCompanyNameBlur = event => {
    if(!this.isCompanyNameValid())
      toast.error("Please enter company name", toastConfig);
  };

  handleWebsiteBlur = event => {
    if(!this.isWebsiteValid())
      toast.error("Please enter a valid website url", toastConfig);
  };

  handleSubmit = event => {
    event.stopPropagation();
    event.preventDefault();

    submitCompanyDetails(this.state.companyName, this.state.website).then(res => {
      console.log(res);
      // TODO: check response before treating it as successfull
      toast.info('Successfull', toastConfig);
    }).catch(err => {
      console.log(err);
      toast.error('Something went wrong!', toastConfig);
    });
  };

  render() {
    const isFormValid = this.isCompanyNameValid() && this.isWebsiteValid();
    console.log(this.state.website, this.isWebsiteValid());
    return (
      <div className="authpage section innerpage">
        <div className="container">
          <div className="wrapper">
            <Animated className="leftwrap center" animationIn="fadeIn" animationOut="fadeOut" isVisible={true}>
              <form className="loginfrm" onSubmit={this.handleSubmit} >
                <h3 className="dashed">Tell us about your company</h3>
                <div className="section-divider-line"></div>
                <div className="frmcntl">
                  <input className="field w-input"
                    name="companyName"
                    value={this.state.companyName}
                    onBlur={this.handleCompanyNameBlur}
                    onChange={this.handleInputChange}
                    placeholder="Company Name"
                  />
                </div>
                <div className="frmcntl">
                  <input
                    className="field w-input"
                    name="website"
                    placeholder="Website"
                    value={this.state.website}
                    onBlur={this.handleWebsiteBlur}
                    onChange={this.handleInputChange}
                  />
                </div>
                <div className="frmcntl">
                  <input className="button submit-button w-button"
                    type="submit"
                    value="Next"
                    disabled={!isFormValid}
                  />
                </div>

                <ToastContainer hideProgressBar={true} />
              </form>
              <div className="support"></div>
            </Animated>
          </div>
        </div>
      </div>
    );
  }
}