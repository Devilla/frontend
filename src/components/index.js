export { default as WebsiteHeader } from "./WebsiteHeader/WebsiteHeader";
export { default as WebsiteFooter } from "./WebsiteFooter/WebsiteFooter";
export { default as WebsiteHome } from "./WebsiteHome/WebsiteHome";
export { default as WebsiteHowItWorks } from "./WebsiteHowItWorks/WebsiteHowItWorks";
export { default as WebsiteIntegrations } from "./WebsiteIntegrations/WebsiteIntegrations";
export { default as WebsitePricing } from "./WebsitePricing/WebsitePricing";
export { default as WebsiteAbout } from "./WebsiteAbout/WebsiteAbout";
export { default as WebsiteSignIn } from "./WebsiteSignIn/WebsiteSignIn";
export { default as WebsiteSignUp } from "./WebsiteSignUp/WebsiteSignUp";
export { default as WebsiteContact } from "./WebsiteContact/WebsiteContact";
export { default as WebsiteTerms } from "./WebsiteTerms/WebsiteTerms";
export { default as WebsitePolicy } from "./WebsitePolicy/WebsitePolicy";
export { default as WebsiteDemoPage}  from "./WebsiteDemoPage/WebsiteDemoPage";
export { default as WebsiteCustStory}  from "./WebsiteCustStory/WebsiteCustStory";
export { default as WebsiteFeature}  from "./WebsiteFeature/WebsiteFeature";
export { default as Affiliate}  from "./Affiliate/Affiliate";
export { default as AffiliateRegister}  from "./Affiliate/AffiliateRegister";

export { default as Header } from "./Header/Header";
export { default as ForgetPassword } from "./ForgetPassword";
export { default as PricePage } from "./PricePage";
export { default as Common } from "./Common/function";
export { default as ResetPassword } from "./ResetPassword";
export { default as Sidebar } from "./Sidebar/Sidebar";
export { default as StatsCard } from "./StatsCard/StatsCard";
export { default as Dashboard } from './Dashboard/Dashboard';
export { default as Notification } from './Notification';
export { default as Analytics } from './Analytics/analytics';
export { default as AnalyticsProfile } from './Analytics/AnalyticsProfile';
export { default as New } from './New/new';
export { default as LoginFlow } from './LoginFlow/LoginFlow';
export { default as Spinner } from './Spinner/Spinner';
export { default as Error } from './Error/Error';
export { default as ComingSoon } from './ComingSoon/ComingSoon';
export { default as Button } from './Button/Button';
export { default as Profile } from './Profile/Profile';
export { default as BillingDetails } from './Billing/BillingDetails';
export { default as UpgradeCard } from './UpgradeCard/UpgradeCard';
export { default as UpgradePlan } from './UpgradePlan/UpgradePlan';
export { default as Integration } from './Integration/Integration';
export { default as Template } from './Template';
export { default as VerificationPage } from './VerificationPage/VerificationPage';
