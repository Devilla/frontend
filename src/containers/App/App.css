@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
