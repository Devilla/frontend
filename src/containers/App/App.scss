@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
