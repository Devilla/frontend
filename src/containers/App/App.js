import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
