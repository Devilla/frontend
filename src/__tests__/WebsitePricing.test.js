import React from 'react'
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer'
import { WebsitePricing } from 'components';
import { store } from 'App';


// Snapshot for WebsitePricing React Component
describe('>>>P R I C I N G --- Snapshot',()=>{
    it('+++capturing Snapshot of pricing', () => {
        const renderedValue =  renderer.create(<WebsitePricing store={store} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});
