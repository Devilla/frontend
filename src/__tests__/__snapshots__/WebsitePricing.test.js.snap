// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>P R I C I N G --- Snapshot +++capturing Snapshot of pricing 1`] = `
<div
  className="main-container"
>
  <section
    className="text-center bg--secondary"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-11 col-lg-10"
        >
          <h2>
            Simple, honest & affordable pricing 
          </h2>
          <p
            className="lead h4"
          >
            Enjoy any plan free for 7 days. Cancel anytime, No questions asked
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className="text-center"
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-center"
      >
        <div
          className="col-md-1 mr-0 text-left "
          id="leftmg"
        >
          <div>
            <strong
              className="h5 type--bold"
              onClick={[Function]}
            >
              Monthly
            </strong>
          </div>
        </div>
        <div
          className="col-md-1 col-sm-1 my-auto text-center pl-2"
        >
          <input
            checked={true}
            className="tgl tgl-ios"
            id="cb2"
            type="checkbox"
          />
          <label
            className="tgl-btn"
            for="cb2"
            onChange={[Function]}
          />
        </div>
        <div
          className="text-left my-auto"
        >
          <strong
            className="h5 type--bold"
            onClick={[Function]}
          >
            Yearly
          </strong>
        </div>
      </div>
    </div>
  </section>
  <section
    className="text-center unpad--top"
  >
    <div
      className="container"
    >
      <div
        className="row"
      />
    </div>
  </section>
  <hr
    className="my-auto col-md-6"
  />
  <section
    className="text-center pt-5 mt-4 mb-5"
  >
    <div
      className="container "
    >
      <div
        className="row"
      >
        <div
          className="col-md-10 col-lg-7"
        >
          <div
            className="cta"
          >
            <span
              className="h2 my-auto pb-5"
            >
              Looking for a Bigger Plan?  
            </span>
            <a
              className="btn btn--primary "
              onClick={[Function]}
              style={Object {}}
            >
              <span
                className="btn__text"
              >
                Contact Us
              </span>
              <br />
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    className="bg--secondary"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-6"
        >
          <div
            className="text-block"
          >
            <h2
              className="pt-3"
            >
              Frequently asked questions
            </h2>
            <h4>
                Need more help? Check out our Help Center 
            </h4>
          </div>
        </div>
        <div
          className="col-lg-6"
        >
          <div
            className="text-block"
          >
            <h4
              className="mb-2 type--bold"
            >
              What do you mean by visitors?
            </h4>
            <p
              className="type--fine-print text-justify  h3"
            >
              The billing counts unique visitors as the metric for billing. A unique visitor is the one who visits the website page where the pixel code is installed. A visitor can visit the page multiple times on all the pixelated pages but that will be counted as one unique visit only.
            </p>
          </div>
          <hr />
          <div
            className="text-block"
          >
            <h4
              className="mb-2 type--bold"
            >
              But will it work for me?
            </h4>
            <p
              className="type--fine-print text-justify h3"
            >
              Influence works for all the websites and platforms that are out there. We are rolling out more and more integrations every month so that you can sync them up with your favorite services and marketing tools. You can even see the integrations we have listed on our integrations section.
            </p>
          </div>
          <hr />
          <div
            className="text-block "
          >
            <h4
              className="mb-2 type--bold"
            >
              Will Influence work for all the customers on my website?
            </h4>
            <p
              className="type--fine-print text-justify h3"
            >
              Influence not only works for customer details capturing, but it also works for lead captures, webinar pages and other places where you can capture the customer’s details.  
            </p>
          </div>
          <hr />
          <div
            className="text-block "
          >
            <h4
              className="mb-2 type--bold"
            >
              Are these notifications legit?
            </h4>
            <p
              className="type--fine-print text-justify h3"
            >
              When we thought of building this product, we thought of helping brands and customers bring transparency to each other in the nicest possible manner. If we get to work together and you use our product, all the notifications that you will see on your website will be 100% legit and the data would be the one which your customers would use.
            </p>
          </div>
          <hr />
          <div
            className="text-block "
          >
            <h4
              className="mb-2 type--bold"
            >
              Can I cancel it anytime?
            </h4>
            <p
              className="type--fine-print tesxt-justify h3"
            >
              Yes. You can cancel the subscription whenever you want. If you are in monthly contract you will only be billed on a monthly basis.
            </p>
          </div>
          <hr />
          <div
            className="text-block"
          >
            <h4
              className="mb-2 type--bold"
            >
              What will happen if I get more unique visitors on plan? Will it charge me automatically?
            </h4>
            <p
              className="type--fine-print text-justify h3"
            >
              Once you start inching closer to your traffic limit, we’ll send you notifications before you even actually hit that limit. Once you go above your plan limit we’ll automatically upgrade you for the next plan after your consent for payment.
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
